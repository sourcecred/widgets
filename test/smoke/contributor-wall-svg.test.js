"use strict";

require = require("esm")(module /*, options*/);

const minScoresV1 = require("./data/min-scores-v1.json");
const {createTmpCacheDir, clearTmpCacheDir} = require("../util/caching");
const {createContributorWall} = require("../../src/svg");

test("Expect min-scores-v1 SVG to match snapshot", async () => {
  // Given
  const users = minScoresV1[1].users;
  const SOURCECRED_GITHUB_TOKEN = null;
  const options = {
    minCred: 0,
    maxUsers: 10,
    usersPerRow: 10,
    avatarSize: 64,
    margin: 5,
  };

  // When
  const svg = await createContributorWall(
    users,
    SOURCECRED_GITHUB_TOKEN,
    options
  );

  // Then
  expect(svg).toMatchSnapshot();
});

test("Expect image caching to produce identical results", async () => {
  // Given
  const users = minScoresV1[1].users;
  const SOURCECRED_GITHUB_TOKEN = null;
  const cacheDir = createTmpCacheDir();
  const options = {
    minCred: 0,
    maxUsers: 10,
    usersPerRow: 10,
    avatarSize: 64,
    margin: 5,
    cacheDir,
  };

  // When
  const svg1 = await createContributorWall(
    users,
    SOURCECRED_GITHUB_TOKEN,
    options
  );
  const svg2 = await createContributorWall(
    users,
    SOURCECRED_GITHUB_TOKEN,
    options
  );

  // Then
  expect(svg1).toEqual(svg2);
  expect(svg1).toMatchSnapshot();

  // Cleanup
  clearTmpCacheDir(cacheDir);
});
