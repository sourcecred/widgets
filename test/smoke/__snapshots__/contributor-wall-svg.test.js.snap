// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect image caching to produce identical results 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" width=\\"685\\" height=\\"64\\">
		<style>.link { cursor: pointer; }</style>
		<a href=\\"https://github.com/credbot\\" class=\\"link\\" target=\\"_blank\\" id=\\"credbot\\" aria-label=\\"Visit the profile of credbot at GitHub\\"><image x=\\"0\\" y=\\"0\\" width=\\"64\\" height=\\"64\\" xlink:href=\\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAIAAAAlC+aJAAAACXBIWXMAAAsSAAALEgHS3X78AAAFzElEQVRo3u1Zy0sbXRSvn9pWDcZqqtRaFHyi+IAu3FkUCoKbWrurBLp14UJ0JT7BqhER9E/QVow2C5FI1CIFsaAgWh+xLUYqClqqYhEfqdbvR370cpn4SPlmBj6YsxjuTCbn/s7znnPm1i2DDDLIIIMMMsggvSkoKOh/jP4fH6krA3nqpHvsFBwcrKIdBHTNZQgNDcW1vLx8eHj4zp07N24Z5KPrX6AuOjs7bTYbFrjVyj9DQkJwffbs2cnJycXFhd1u5xPFfrSP4iGBKqQV6Gtray98VF9fr5UM3LusrOz09BQ78ep0Ok0mE3Hgevv2bYqk8DcFH7zGEKI929vbweqXj7BobGxUP0kQxMuXL6n7379/43p2doYrfAmgZaDZ2dnV1dWDg4PT09OLi4tLS0szMzMjIyPNzc0FBQVAT4YUtampCUzOz89pAS5aW1s1SZoOhwPcvV7vxR+izvAc8QBAL168mJqa4gsU0p/cbndlZaXFYgFDKJtMxMtUypcvX+7du6emHcgIEAcGBi6VweVyffjwgbhBv/7QmY/ELX/Fa58/f3779i0RC/RkBYs9ePBAfS8iOzgA/F5sJtudC7G+igBXvCNbibr3eDxJSUla5VMyDQsLg0OLLcX28i2QyapVaJqiyirgf2GWlJQUZiFtKwhkD8SorHsZmSyJ4lYhp2zA5eXl+Ph4Pc4yZkDEA/xeYQfhHojC169fFxcXZ2ZmpqenP3nyBHnp48eP/FUWm+tv374lJiZqq3uFEXB9//69LAAXP378sFqtTPD+/0IaRWKVQ4gCbG1txcbG6lQjUknImP75e2NjA4eAOIxpKx4RsBh9w2w2T0xM+Eve0dGhqwUmJyfF3ozO/f39nJwcZqqrFMnDKzIycnZ2VojNv+/s7Ny/f19zI1CLubm59AHmdeJ49eqVgHijAcHB/1APkIMK9RwiUvgx0a+srNy9ezdA/VGGN2/eCOi8vnv3TvMsRO5DQ0NiV4qBnBO48ihAaWmpkJ/X1dVVdduMKwVAQhQCcO+SkpLAQ5D44uLifv78KSff79+/46EeuYipELvSg7HIysoK3PrEh/rv69evMp+jo6PU1FQ9zjLUW/LGMEVGRsbfCoBkBZ+R+RwfH6elpWkrAFkzCco5tLCw8G9dKCoqand3V6QyLHD78OFDPQSQizkGcU1Nzd8GcVFRkZCfMbC2tsY+W8MY4N4NDQ0COsVAHxP4xpSzu7tbwWR8fFzzAKAAKGmE8oQDPH36NBAvIj64CqomwYRi4HjR/CAT5fTCwoIii8/Pz0dHR1+PQAyUent7FVGEgxl1qx4piPgqKir8CzLUpyaTiXZQjEbkWUtXV5dcCPK/6DD1QC+MgKYbDZSMg24wNzeXl5cnXhNDSN6i2bXb7bLkIoHm5+frJwCN0N/fr2hoKAzQoPd//vw5Oqzw8HC0oBaLBTmnp6cHZ+2lDQ1KqYiICJ0EoBPX1dVd1VKKh4eHh2jN3G733t6e3Ctf2lKikrt0zqey7ome8xyF7uXm5tRHchfv9XoRprLnKHplyhDIvPW/om9ra1NMGUQmVYwexE+Kh6J2kJ8zhJxOJ0d3Ws1G6TnyLI1DLhyim5ubBAdNQ/0QhkBZbOJWNsLBwQErQnlGxjVqdfVlIC+0rQrPodoAJSEhASFrs9m2t7evmivyIQKjr6/v8ePHaIDgM4oZGZmPjY0h9NUfLba0tPjnflSmjx49EiaKjY21Wq2Dg4OoNCEM4KJIxqELEwFWVVVVcnKy+G6AqFXIQJ6otTjXUNMIxCeqIG7p8XgICPshQuR2HmtkfZTHOF8hIfKp/JUE0Dm2wPPR0VGZp8Ph0CoM5EoO9OnTJ+peUf9c9XmCKpd/EoUJe1Si1zCZikQEX1pfX4ffX1O9iU9pYkB0zZwPV5fLBVOIW82/lMXExKilJzIxm82RkZE6DedU15MY4On0+VkLK9/4PdMggwwyyCCDDDJIA/oXekBvQiw5MFsAAAAASUVORK5CYII=\\" /></a>
	</svg>"
`;

exports[`Expect min-scores-v1 SVG to match snapshot 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" width=\\"685\\" height=\\"64\\">
		<style>.link { cursor: pointer; }</style>
		<a href=\\"https://github.com/credbot\\" class=\\"link\\" target=\\"_blank\\" id=\\"credbot\\" aria-label=\\"Visit the profile of credbot at GitHub\\"><image x=\\"0\\" y=\\"0\\" width=\\"64\\" height=\\"64\\" xlink:href=\\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEAAAABACAIAAAAlC+aJAAAACXBIWXMAAAsSAAALEgHS3X78AAAFzElEQVRo3u1Zy0sbXRSvn9pWDcZqqtRaFHyi+IAu3FkUCoKbWrurBLp14UJ0JT7BqhER9E/QVow2C5FI1CIFsaAgWh+xLUYqClqqYhEfqdbvR370cpn4SPlmBj6YsxjuTCbn/s7znnPm1i2DDDLIIIMMMsggvSkoKOh/jP4fH6krA3nqpHvsFBwcrKIdBHTNZQgNDcW1vLx8eHj4zp07N24Z5KPrX6AuOjs7bTYbFrjVyj9DQkJwffbs2cnJycXFhd1u5xPFfrSP4iGBKqQV6Gtray98VF9fr5UM3LusrOz09BQ78ep0Ok0mE3Hgevv2bYqk8DcFH7zGEKI929vbweqXj7BobGxUP0kQxMuXL6n7379/43p2doYrfAmgZaDZ2dnV1dWDg4PT09OLi4tLS0szMzMjIyPNzc0FBQVAT4YUtampCUzOz89pAS5aW1s1SZoOhwPcvV7vxR+izvAc8QBAL168mJqa4gsU0p/cbndlZaXFYgFDKJtMxMtUypcvX+7du6emHcgIEAcGBi6VweVyffjwgbhBv/7QmY/ELX/Fa58/f3779i0RC/RkBYs9ePBAfS8iOzgA/F5sJtudC7G+igBXvCNbibr3eDxJSUla5VMyDQsLg0OLLcX28i2QyapVaJqiyirgf2GWlJQUZiFtKwhkD8SorHsZmSyJ4lYhp2zA5eXl+Ph4Pc4yZkDEA/xeYQfhHojC169fFxcXZ2ZmpqenP3nyBHnp48eP/FUWm+tv374lJiZqq3uFEXB9//69LAAXP378sFqtTPD+/0IaRWKVQ4gCbG1txcbG6lQjUknImP75e2NjA4eAOIxpKx4RsBh9w2w2T0xM+Eve0dGhqwUmJyfF3ozO/f39nJwcZqqrFMnDKzIycnZ2VojNv+/s7Ny/f19zI1CLubm59AHmdeJ49eqVgHijAcHB/1APkIMK9RwiUvgx0a+srNy9ezdA/VGGN2/eCOi8vnv3TvMsRO5DQ0NiV4qBnBO48ihAaWmpkJ/X1dVVdduMKwVAQhQCcO+SkpLAQ5D44uLifv78KSff79+/46EeuYipELvSg7HIysoK3PrEh/rv69evMp+jo6PU1FQ9zjLUW/LGMEVGRsbfCoBkBZ+R+RwfH6elpWkrAFkzCco5tLCw8G9dKCoqand3V6QyLHD78OFDPQSQizkGcU1Nzd8GcVFRkZCfMbC2tsY+W8MY4N4NDQ0COsVAHxP4xpSzu7tbwWR8fFzzAKAAKGmE8oQDPH36NBAvIj64CqomwYRi4HjR/CAT5fTCwoIii8/Pz0dHR1+PQAyUent7FVGEgxl1qx4piPgqKir8CzLUpyaTiXZQjEbkWUtXV5dcCPK/6DD1QC+MgKYbDZSMg24wNzeXl5cnXhNDSN6i2bXb7bLkIoHm5+frJwCN0N/fr2hoKAzQoPd//vw5Oqzw8HC0oBaLBTmnp6cHZ+2lDQ1KqYiICJ0EoBPX1dVd1VKKh4eHh2jN3G733t6e3Ctf2lKikrt0zqey7ome8xyF7uXm5tRHchfv9XoRprLnKHplyhDIvPW/om9ra1NMGUQmVYwexE+Kh6J2kJ8zhJxOJ0d3Ws1G6TnyLI1DLhyim5ubBAdNQ/0QhkBZbOJWNsLBwQErQnlGxjVqdfVlIC+0rQrPodoAJSEhASFrs9m2t7evmivyIQKjr6/v8ePHaIDgM4oZGZmPjY0h9NUfLba0tPjnflSmjx49EiaKjY21Wq2Dg4OoNCEM4KJIxqELEwFWVVVVcnKy+G6AqFXIQJ6otTjXUNMIxCeqIG7p8XgICPshQuR2HmtkfZTHOF8hIfKp/JUE0Dm2wPPR0VGZp8Ph0CoM5EoO9OnTJ+peUf9c9XmCKpd/EoUJe1Si1zCZikQEX1pfX4ffX1O9iU9pYkB0zZwPV5fLBVOIW82/lMXExKilJzIxm82RkZE6DedU15MY4On0+VkLK9/4PdMggwwyyCCDDDJIA/oXekBvQiw5MFsAAAAASUVORK5CYII=\\" /></a>
	</svg>"
`;
